### PR TITLE
tools: improvements to reduce complexity.

### PR DESCRIPTION
**Pull request by lgirdwood**
Reduce development complexity by
1) removing deprecated fuzzer.
2) parse all topology objects in memory rather than by seeking file, reading into allocated memory and then freeing memory.

### CODE REVIEW

**Comment by lgirdwood** [comment]
@singalsu pls see - this should simplify further testbench and tplg parser updates.

File: tools/testbench/testbench.c [hunk 1]
```diff
@@ -649,7 +649,7 @@ static int test_pipeline_load(struct pipeline_thread_data *ptdata, struct tplg_c
 	memset(ctx, 0, sizeof(*ctx));
 	ctx->comp_id = 1000 * ptdata->core_id;
 	ctx->core_id = ptdata->core_id;
-	ctx->file = tp->file;
+	//ctx->file = tp->file;
```
Comment (singalsu) [inline tools/testbench/testbench.c:652]: Not finished cleanup?

Comment (lgirdwood) [reply tools/testbench/testbench.c]: Thanks, fixed.

**Review by singalsu — APPROVED** [review]
Thanks! Looks good, also there only a small easy to fix conflict fix with static tesbench PR.
There's a small typo to fix cant -> can't found by checkpatch.

This would be good to merge before static testbench.

**Comment by lgirdwood** [comment]
@kv2019i pls review.

File: tools/testbench/topology.c [hunk 2]
```diff
@@ -767,6 +725,7 @@ static int load_filewrite(struct tplg_context *ctx, int dir)
 	register_comp(filewrite.comp.type, NULL);
 	if (ipc_comp_new(sof->ipc, ipc_to_comp_new(&filewrite)) < 0) {
 		fprintf(stderr, "error: new file write\n");
+		free(filewrite.fn);
```
Comment (kv2019i) [inline tools/testbench/topology.c:728]: This took a while to understand, but seems to be a drive-by bugfix to an unrelated bug in the function.

**Review by kv2019i — APPROVED** [review]

File: tools/tplg_parser/tokens.c [hunk 3]
```diff
@@ -334,5 +335,9 @@ bool is_valid_priv_size(size_t size_read, size_t priv_size,
 	 * check if size of data to be read from widget's private data
 	 * doesn't exceed private data's size.
 	 */
-	return size_read + arr_size + arr_elems_size <= priv_size;
+	valid = size_read + arr_size + arr_elems_size <= priv_size;
```
[the remark this answers is not in the document]
Comment (lgirdwood) [reply tools/tplg_parser/tokens.c:338]: Thanks @marc-hb will take this as an incremental, there is more pending after this.
